src/uct/ib/rc/base/rc_ep.h

```diff
@@ -185,7 +193,7 @@ static inline void uct_rc_ep_process_tx_completion(uct_rc_iface_t *iface,
 
     ucs_queue_for_each_extract(op, &ep->outstanding, queue,
                                UCS_CIRCULAR_COMPARE16(op->sn, <=, sn)) {
-        op->handler(op);
+        op->handler(op, UCS_OK);
```

Comment:
We get into this function if completion status is ok only. As soon as failure occurs we handle it in uct_rc_verbs_handle_failure function (where we "remove" ep from the iface lookup array). Thus we will skip all completions of the failed endpoint since failure has been detected.